modules/react/select/stories/examples/FetchingDynamicItems.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+import {FormField} from '@workday/canvas-kit-react/form-field';
+import {Select, useSelectModel} from '@workday/canvas-kit-react/select';
+import {Flex} from '@workday/canvas-kit-react/layout';
+
+export const FetchingDynamicItems = () => {
+  const [value, setValue] = React.useState('John Wick');
+  const [moviesLists, setMoviesList] = React.useState([]);
+  const [isLoading, setIsLoading] = React.useState(false);
+
+  React.useEffect(() => {
+    setTimeout(() => {
+      setIsLoading(true);
+      fetch('https://my-json-server.typicode.com/horizon-code-academy/fake-movies-api/movies')
+        .then(response => response.json())
+        .then(data => {
+          setMoviesList(data);
+        });
+    }, 2000);
+    setIsLoading(false);
```

Comment:
Wrapped this in a set timeout to make sure the select doesn't crash waiting for items

Reply:
That doesn't make sense to me. This seems backwards. `isLoading` is false while we're loading, and `true` when we're done loading.

Is the `setIsLoading(false)` necessary if the initial value is already `false`?

Also, we have a `useMount` hook from common for this type of use-case. Otherwise there's ESLint rules that trip because of the dependency list. An empty array in the `useEffect` dependency array means "there are no dependencies, only run this on initial render". The `useMount` has a much clearer intent in the name.

Also, I recommend against using `fetch` for any external API. It is common to have a Content Security Policy on the server to only allow JavaScript/images/etc to communicate with certain URLs. Locally, this doesn't apply, but can cause problems with Chromatic, or any other place this Storybook will be loaded (possibly even the Canvas Site).

I have loading examples like this that use `setTimeout` instead. Also, fetch is generally not good to use without some type of wrapper since the fetch could resolve after the component is unmounted. This causes the React warning about calling a state setter after the component is unmounted. The `setTimeout` returns an integer that can be passed to `clearTimeout` on unmount to unsubscribe the callback so the setter isn't called when the component unmounts.